plugins/woocommerce/client/blocks/assets/js/blocks/payment-method-icons/edit.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, RangeControl } from '@wordpress/components';
+import { getPaymentMethods } from '@woocommerce/blocks-registry';
+
+const CardPreview = ( {
+	type,
+	icon,
+}: {
+	type: string | undefined;
+	icon: string | undefined;
+} ) => {
+	let CardIcon = null;
+
+	if ( type && icon ) {
+		CardIcon = (
+			<div className="wp-block-woocommerce-payment-method-icons__item">
+				<span
+					className="wp-block-woocommerce-payment-method-icons__icon"
+					style={ {
+						backgroundImage: `url(${ icon })`,
+					} }
+					role="img"
+					aria-label={ type }
+				/>
+			</div>
+		);
+	}
+
+	return CardIcon;
+};
+
+const Edit = ( {
+	attributes,
+	setAttributes,
+}: {
+	attributes: {
+		numberOfIcons: number;
+	};
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
```

Comment:
Minor but you can make it more strict by extracting attributes type and reusing it in `setAttributes`

```suggestion
type BlockAttributes = {
  numberOfIcons: number;
}

...

	attributes: BlockAttributes,
	setAttributes: ( attributes: Partial<BlockAttributes> ) => void;
```

Reply:
Nice, thank you! Done in https://github.com/woocommerce/woocommerce/pull/57181/commits/e10e1bc4d3da0cb727ccce091d40818f7e21211c.